stress/common/src/main/java/alluxio/stress/common/FileSystemParameters.java

```diff
@@ -33,9 +35,17 @@
       converter = FileSystemParameters.FileSystemParametersReadTypeConverter.class)
   public ReadType mReadType = ReadType.CACHE;
 
+  @Parameter(names = {"--write-type"},
+      description = "The write type to use when creating files. Options are [MUST_CACHE, "
+          + "CACHE_THROUGH, THROUGH, ASYNC_THROUGH, ALL]",
+      converter = FileSystemParameters.FileSystemParametersWriteTypeConverter.class)
+  public String mWriteType = InstancedConfiguration.defaults()
```

Comment:
@shiqi1122 If I do `alluxio runClass --write-type MUST_CACHE --conf blah.writetype=CACHE_THROUGH`, `MUST_CACHE` will be respected. But if I do `alluxio runClass  --conf blah.writetype=CACHE_THROUGH`, this write type will default to the configuration property default ASYNC_THROUGH.

I think the problem here is this param respects the Alluxio property (`InstancedConfig.default()` merges from alluxio-site.properties and java opts and so on but it does not give you what's passed from `--conf`. 

There are two ways to implement the fix, up to you which one is easier in implementation:

Option 1: Still default to `InstancedConfiguration.defaults()`. At runtime, check `--conf` and look for write type settings. If there's one write type setting and if it has a different value from the `mWriteType`, throw an error and quit. The backward compatibility is slightly worse, because if a user sets `--conf writetype=CACHE_THROUGH` but not this `--write-type`, the `mWriteType` will be ASYNC_THROUGH and throw an error. However this is acceptable.

Option 2: Respect the `--conf`. If `--write-type` and `--conf writetype` do not agree, still throw an error. But if `--write-type` is not set, take from `--conf writetype`. This has high backward compatibility. But it's not good for the users to rely on old properties so it's not completely good.

Reply:
After some more thoughts I prefer Option 1. 

Our intention of adding this flag is to make overriding the write type easy. If the user has config properties like `alluxio.user.file.writetype.default=CACHE_THROUGH` in worker nodes, we want to still be able to test `MUST_CACHE` without having to update the config files. We want to also be more sure when we set `--write-type` to a certain value and be confident that is the real write type used (not overriden by anything else). 

By Option 1, this `--write-type` flag is only controlled by our command. And if (for any possible reason) this `--conf` has a different value, or the configuration files have a different value, we should make sure `--write-type` is used.

So I think what you can do is to compare the `--write-type` with the `--conf` and with the `InstancedConfiguration.defaults()`. If they do not match, print a warning or something and let the user know that `--write-type` value is used.